src/wallet/wallet_rpc_server.cpp

```diff
@@ -615,22 +624,47 @@ namespace tools
         return false;
       }
 
-      if (!req.do_not_relay)
-        m_wallet->commit_tx(ptx_vector);
-
-      // populate response with tx hash
-      res.tx_hash = epee::string_tools::pod_to_hex(cryptonote::get_transaction_hash(ptx_vector.back().tx));
-      if (req.get_tx_key)
+      if (m_wallet->multisig())
       {
-        res.tx_key = epee::string_tools::pod_to_hex(ptx_vector.back().tx_key);
+        res.multisig_txset = epee::string_tools::buff_to_hex_nodelimer(m_wallet->save_multisig_tx(ptx_vector));
+        if (res.multisig_txset.empty())
+        {
+          er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+          er.message = "Failed to save multisig tx set after creation";
+          return false;
+        }
       }
-      res.fee = ptx_vector.back().fee;
-
-      if (req.get_tx_hex)
+      else
       {
-        cryptonote::blobdata blob;
-        tx_to_blob(ptx_vector.back().tx, blob);
-        res.tx_blob = epee::string_tools::buff_to_hex_nodelimer(blob);
+        if (!req.do_not_relay)
+          m_wallet->commit_tx(ptx_vector);
+
+        // populate response with tx hash
+        res.tx_hash = epee::string_tools::pod_to_hex(cryptonote::get_transaction_hash(ptx_vector.back().tx));
+        if (req.get_tx_key)
+        {
+          res.tx_key = epee::string_tools::pod_to_hex(ptx_vector.back().tx_key);
```

Comment:
Due to the subaddress scheme, we should also deal with the additional tx keys. In #2487 I proposed to concatenate all the additional tx keys into a single string:

```c++
for (const crypto::secret_key& additional_tx_key : ptx_vector.back().additional_tx_keys)
  res.tx_key += epee::string_tools::pod_to_hex(additional_tx_key);
```
https://github.com/monero-project/monero/pull/2487/files#diff-297e4d3d4221d9bf7ea37267597959f9R626

Reply:
Concatenating seems like a good idea :)